drivers/dma/dma_mcux_edma.c

```diff
@@ -18,6 +18,12 @@
 #include <drivers/dma.h>
 #include <drivers/clock_control.h>
 
+#ifdef CONFIG_HAS_MCUX_CACHE
+#include <cache.h>
+#include <fsl_cache.h>
+#include <linker/linker-defs.h>
```

Comment:
These two includes are no longer needed with your changes, since you removed the cache APIs from this driver.
#include <cache.h>
#include <fsl_cache.h>

Below header will not be needed if removing the check of buffers being in cacheable memory
#include <linker/linker-defs.h>